notifications_utils/formatters.py

```diff
@@ -338,6 +338,9 @@ def strip_all_whitespace(value, extra_characters=''):
 
 
 def strip_and_remove_obscure_whitespace(value):
+    if value == '':
```

Comment:
I did wonder about whether using `re.sub` might be faster here, instead of doing a separate `replace` for each character. It's no match for an equality check, and it turns out it's actually worse than simple iteration:

- iterated `.replace`: 17.9s
- `re.sub`: 18.5s